libp2p/protocols/connectivity/autonatv2/types.nim

```diff
@@ -80,6 +91,12 @@ type
     of MsgType.DialDataResponse:
       dialDataResp*: DialDataResponse
 
+# codec to string
+proc `$`*(self: AutonatV2Codec): string =
+  case self
+  of AutonatV2Codec.DialRequest: "/libp2p/autonat/2/dial-request"
+  of AutonatV2Codec.DialBack: "/libp2p/autonat/2/dial-back"
+
```

Comment:
No need to define this. If you print an AutonatV2Codec enum, it will print the string value you setup in the enum definition
```suggestion
```